jetty-deploy/src/main/java/org/eclipse/jetty/deploy/providers/WebAppProvider.java

```diff
@@ -492,6 +477,16 @@ protected void fileRemoved(String filename) throws Exception
         //is the file that was removed an .xml file?
         if (lowname.endsWith(".xml"))
             super.fileRemoved(filename);
+
+        //is there a .xml file of the same name?
+        if (exists(file.getName()+".xml")||exists(file.getName()+".XML"))
```

Comment:
What if its `.Xml` ?

Reply:
You could use the test from line 478 here, no?

Reply:
Looks like the WebAppProvider has never supported files with extensions of .War or .Xml (see the fileAdded() method).  Not sure we need to start now.